### PR TITLE
Mb add empty graph element

### PR DESCRIPTION
**Pull request by Mbuckley0**

### CODE REVIEW

File: graphs/DataVisualisation/SingleNumber/index.js [hunk 1]
```diff
@@ -31,12 +34,15 @@ module.exports = React.createClass({
   },
 
   render: function() {
-    var loading = this.props.loading,
+    var loading    = this.props.loading,
+        emptyState = this.props.emptyState,
         className;
-
     if (loading) {
```
Comment (davidherse) [inline graphs/DataVisualisation/SingleNumber/index.js]: You could make this neater with the classname module.
`var classnames = require('classnames');`
```js
var classes = classnames({
      "hui-SingleNumber--loading": loading,
      "hui-SingleNumber--emptyState": emptyState
    }, "hui-SingleNumber");
```

File: graphs/DataVisualisation/index.js [hunk 2]
```diff
@@ -56,9 +59,15 @@ module.exports = React.createClass({
...
+        loading        = props.loading,
+        emptyState     = props.emptyState;
 
-    if (loading || typeof(total) != 'undefined') {
+    if (emptyState) {
+      return (
+        <SingleNumber title={ title } format={ totalFormat } value={ valueConverter(total) } emptyState={ emptyState }/>
+      );
+    }
+    else if (loading || typeof(total) != 'undefined') {
```
Comment (davidherse) [inline graphs/DataVisualisation/index.js]: `!==`

File: graphs/DataVisualisation/index.js [hunk 3]
```diff
@@ -66,21 +75,26 @@ module.exports = React.createClass({
...
+    var props      = this.props,
+        delta      = props.delta,
+        loading    = props.loading,
+        emptyState = props.emptyState;
 
-    if (loading || delta) {
+    if (emptyState) {
+      return <DeltaArrow delta={ delta } emptyState={ emptyState } />;
+    }
+    else if (loading || delta) {
```
Comment (davidherse) [inline graphs/DataVisualisation/index.js]: You don't need the else if here. The return in the above statement mean you will never reach this point if the first if is true.

File: graphs/DeltaArrow/index.js [hunk 4]
```diff
@@ -9,22 +9,25 @@ module.exports = React.createClass({
...
-        delta   = props.delta,
-        loading = props.loading,
+    var props      = this.props,
+        delta      = props.delta,
+        loading    = props.loading,
+        emptyState = props.emptyState,
         path;
 
-    if (delta == null || delta > 0 || loading == true) {
+    if (delta == null || delta > 0 || loading == true || emptyState == true) {
```
Comment (davidherse) [inline graphs/DeltaArrow/index.js]: Maybe move this into a var to make it a bit clearer. 
`var showBlankState = (delta === null || delta > 0 || loading === true || emptyState === true);`

Also you should use `===`. It's more equal. Look it up.

File: graphs/DeltaArrow/index.js [hunk 5]
```diff
@@ -38,14 +41,17 @@ module.exports = React.createClass({
...
-        loading = props.loading,
+    var props      = this.props,
+        delta      = props.delta,
+        loading    = props.loading,
+        emptyState = props.emptyState,
         className,
         text;
 
-    if (loading == true ) {
+    if (loading == true) {
```
Comment (davidherse) [inline graphs/DeltaArrow/index.js]: Use the classname module to build classes.

File: graphs/DataVisualisation/index.js (hunk 3 above)
Comment (Mbuckley0) [inline graphs/DataVisualisation/index.js]: I think that if it doesn't meet any of the conditions emptyState, loading or delta then it is not supposed to render a DeltaArrow.

File: graphs/DeltaArrow/index.js [hunk 6]
```diff
@@ -38,14 +41,17 @@ module.exports = React.createClass({
...
       className = "hui-DeltaArrow--loading";
+    } else if (emptyState == true) {
+      className = "hui-DeltaArrow--emptyState";
     } else if (delta == 0) {
```
Comment (davidherse) [inline graphs/DeltaArrow/index.js]: We should mix logic that is build classnames and controlling render. Also avoid long if else statements. It smells.

File: graphs/DataVisualisation/index.js (hunk 3 above)
Comment (davidherse) [inline graphs/DataVisualisation/index.js]: Maybe. At least we should take out the `else`. It's not needed.

File: graphs/DataVisualisation/index.js [hunk 7]
```diff
@@ -56,9 +59,15 @@ module.exports = React.createClass({
...
+        loading        = props.loading,
+        emptyState     = props.emptyState;
 
-    if (loading || typeof(total) != 'undefined') {
+    if (emptyState) {
+      return (
+        <SingleNumber title={ title } format={ totalFormat } value={ valueConverter(total) } emptyState={ emptyState }/>
+      );
+    }
+    else if (loading || typeof(total) !== 'undefined') {
```
Comment (davidherse) [inline graphs/DataVisualisation/index.js]: You still don't need this else. Just the if. The above if will return the function early if true.

File: graphs/LineGraph/index.js [hunk 8]
```diff
@@ -179,8 +183,16 @@ module.exports = React.createClass({
       graph = this.renderGraph();
     }
 
+    if(emptyState === true) {
+      emptyData =
```
Comment (davidherse) [inline graphs/LineGraph/index.js]: Brackets make it clearer.
```js
emptyData =  (
   <span className="hui-LineGraph__emptyState">
      No Information to Display
   </span>);
```

Comment (Mbuckley0) [inline graphs/LineGraph/index.js]: Okay all fixed.

**Pull request merged**